vscode-lean4/src/utils/leanInstaller.ts

```diff
@@ -13,12 +13,14 @@ export class LeanInstaller implements Disposable {
     private subscriptions: Disposable[] = [];
     private prompting : boolean = false;
     private pkgService : LeanpkgService;
+    private defaultToolchain : string;
 
     private installChangedEmitter = new EventEmitter<string>();
     installChanged = this.installChangedEmitter.event
 
-    constructor(outputChannel: OutputChannel, localStorage : LocalStorageService, pkgService : LeanpkgService){
+    constructor(outputChannel: OutputChannel, localStorage : LocalStorageService, pkgService : LeanpkgService, defaultToolchain : string) {
```

Comment:
While I initially thought the `defaultToolchain` was unused,  I see now that is not that case (and somewhat get what is going on). However, I feel that this a bit confusing that the installer is initialized with a `defaultToolchain` but many of its methods ask for a different `defaultToolchain`.

Reply:
Also, is there any reason why it is not just using the `defaultToolchain` from the `LeanpkgService`?

Reply:
You are right, installElan should be a private method and it no longer needs a defaultToolchain  argument.  So I removed it.